src/prismo/pl/_plotting.py

```diff
@@ -965,3 +997,39 @@ def factor_sparsity_histogram(model: PRISMO, bins: int = 50, nrow: int | None =
     return _plot_sparse_probabilities_histogram(
         model.get_sparse_factor_probabilities("numpy"), bins=bins, nrow=nrow, ncol=ncol
     )
+
+
+def _prettify_factor_names(
+    factors: pd.Series,
+    replacements: dict[str, str] = None,
+    abbreviation_length: int = 40,
+    end_with_database: bool = True,
+) -> pd.Series:
+    """Prettify factor names.
+
+    Replace underscores with spaces. If `replacements` is not empty, replace the keys with the values.
+    Shorten the labels to `abbreviation_length` characters.
+
+    Args:
+        factors: The factor names to prettify.
+        replacements: A dictionary of replacements to make.
+        abbreviation_length: The maximum length of the labels.
+        end_with_database: Whether to end the labels with the database name.
+    """
+    if not isinstance(abbreviation_length, int) or abbreviation_length <= 0:
+        raise ValueError("`abbreviation_length` must be a positive integer.")
+
+    # Replace whitespace
+    factors = factors.str.replace("_", " ", regex=False)
+    # Replace dict values
+    if replacements is not None:
+        for key, value in replacements.items():
+            factors = factors.str.replace(key, value, regex=False)
+    # Shorten string length
+    factors = factors.str.slice(0, abbreviation_length)
```

Comment:
I'd also add an ellipsis (`…`) at the end to make it clear that this is not the full name. One might also think about eliding in the middle of the string, e.g. https://www.xormedia.com/string-truncate-middle-with-ellipsis/